nvtabular/ops/bucketize.py

```diff
@@ -54,15 +56,15 @@ def __init__(self, boundaries):
         super().__init__()
 
     @annotate("Bucketize_op", color="darkgreen", domain="nvt_python")
-    def transform(self, columns: ColumnNames, gdf: cudf.DataFrame) -> cudf.DataFrame:
+    def transform(self, columns: ColumnNames, df: DataFrameType) -> DataFrameType:
         boundaries = {name: self.boundaries(name) for name in columns}
-        new_gdf = cudf.DataFrame()
+        new_df = type(df)()
         for col, b in boundaries.items():
-            # TODO: should just be using cupy.digitize but it's not in 7.8
-            val = 0
-            for boundary in b:
-                val += (gdf[col] >= boundary).astype("int")
-            new_gdf[col] = val
-        return new_gdf
+            new_df[col] = np.digitize(
+                df[col].values,
+                _array(b, like_df=df),
+                right=False,
+            )
+        return new_df
```

Comment:
The deleted comment says that `cupy.digitize` was not supported in 7.8.  However, it looks like [`digitize` was added](https://github.com/cupy/cupy/pull/2758) before that release.  Therefore, I changed the logic to use `np.digitize` (which dispatches to cupy when the data is a cupy array).  I still need to figure out if we need a version check here.

Reply:
Okay - I added a `LooseVersion` check that cupy is >=8.0.0 before using digitize (I confirmed that this is the earliest version with support)